core/arch/arm/kernel/ldelf_loader.c

```diff
@@ -0,0 +1,321 @@
+			n++;
+		}
+	}
+
+	arg->is_arm32 = utc->is_32bit;
+#ifdef ARM32
+		arg->arm32.regs[0] = tsd->abort_regs.r0;
+		arg->arm32.regs[1] = tsd->abort_regs.r1;
+		arg->arm32.regs[2] = tsd->abort_regs.r2;
+		arg->arm32.regs[3] = tsd->abort_regs.r3;
+		arg->arm32.regs[4] = tsd->abort_regs.r4;
+		arg->arm32.regs[5] = tsd->abort_regs.r5;
+		arg->arm32.regs[6] = tsd->abort_regs.r6;
+		arg->arm32.regs[7] = tsd->abort_regs.r7;
+		arg->arm32.regs[8] = tsd->abort_regs.r8;
+		arg->arm32.regs[9] = tsd->abort_regs.r9;
+		arg->arm32.regs[10] = tsd->abort_regs.r10;
+		arg->arm32.regs[11] = tsd->abort_regs.r11;
+		arg->arm32.regs[12] = tsd->abort_regs.ip;
+		arg->arm32.regs[13] = tsd->abort_regs.usr_sp; /*SP*/
+		arg->arm32.regs[14] = tsd->abort_regs.usr_lr; /*LR*/
+		arg->arm32.regs[15] = tsd->abort_regs.elr; /*PC*/
+#endif /*ARM32*/
+#ifdef ARM64
+	if (utc->is_32bit) {
+		arg->arm32.regs[0] = tsd->abort_regs.x0;
+		arg->arm32.regs[1] = tsd->abort_regs.x1;
+		arg->arm32.regs[2] = tsd->abort_regs.x2;
+		arg->arm32.regs[3] = tsd->abort_regs.x3;
+		arg->arm32.regs[4] = tsd->abort_regs.x4;
+		arg->arm32.regs[5] = tsd->abort_regs.x5;
+		arg->arm32.regs[6] = tsd->abort_regs.x6;
+		arg->arm32.regs[7] = tsd->abort_regs.x7;
+		arg->arm32.regs[8] = tsd->abort_regs.x8;
+		arg->arm32.regs[9] = tsd->abort_regs.x9;
+		arg->arm32.regs[10] = tsd->abort_regs.x10;
+		arg->arm32.regs[11] = tsd->abort_regs.x11;
+		arg->arm32.regs[12] = tsd->abort_regs.x12;
+		arg->arm32.regs[13] = tsd->abort_regs.x13; /*SP*/
+		arg->arm32.regs[14] = tsd->abort_regs.x14; /*LR*/
+		arg->arm32.regs[15] = tsd->abort_regs.elr; /*PC*/
+	} else {
+		arg->arm64.fp = tsd->abort_regs.x29;
+		arg->arm64.pc = tsd->abort_regs.elr;
+		arg->arm64.sp = tsd->abort_regs.sp_el0;
+	}
+#endif /*ARM64*/
+
+	res = thread_enter_user_mode((vaddr_t)arg, 0, 0, 0,
+				     usr_stack, utc->dump_entry_func,
+				     is_arm32, &panicked, &panic_code);
+
+#ifdef CFG_WITH_VFP
+	thread_user_clear_vfp(&utc->uctx.vfp);
+#endif
+
+	if (panicked) {
+		utc->dump_entry_func = 0;
+		EMSG("ldelf dump function panicked");
+		abort_print_current_ta();
+		res = TEE_ERROR_TARGET_DEAD;
+	}
+
+	return res;
+}
+
+#ifdef CFG_FTRACE_SUPPORT
+TEE_Result ldelf_dump_ftrace(struct user_ta_ctx *utc, void *buf, size_t *blen)
+{
+	uaddr_t usr_stack = utc->ldelf_stack_ptr;
+	TEE_Result res = TEE_SUCCESS;
+	uint32_t panic_code = 0;
+	uint32_t panicked = 0;
+	size_t *arg = NULL;
+
+	if (!utc->ftrace_entry_func)
+		return TEE_ERROR_NOT_SUPPORTED;
+
+	usr_stack -= ROUNDUP(sizeof(*arg), STACK_ALIGNMENT);
+	arg = (size_t *)usr_stack;
+
+	res = vm_check_access_rights(&utc->uctx,
+				     TEE_MEMORY_ACCESS_READ |
+				     TEE_MEMORY_ACCESS_ANY_OWNER,
+				     (uaddr_t)arg, sizeof(*arg));
+	if (res) {
+		EMSG("ldelf stack is inaccessible!");
+		return res;
+	}
+
+	*arg = *blen;
+
+	res = thread_enter_user_mode((vaddr_t)buf, (vaddr_t)arg, 0, 0,
+				     usr_stack, utc->ftrace_entry_func,
+				     is_arm32, &panicked, &panic_code);
+
+#ifdef CFG_WITH_VFP
```

Comment:
No need for ifdef, `thread_user_clear_vfp()` does nothing unless `CFG_WITH_VFP` defined.

Reply:
Fixed.